features/netsocket/NetworkInterfaceDefaults.cpp

```diff
@@ -110,27 +110,8 @@ MBED_WEAK NetworkInterface *NetworkInterface::get_target_default_instance()
 #elif MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == CELLULAR
 MBED_WEAK NetworkInterface *NetworkInterface::get_target_default_instance()
 {
-    CellularBase *cellular = CellularBase::get_default_instance();
-    if (!cellular) {
-        return NULL;
-    }
-    /* CellularBase is expected to attempt to work without any parameters - we
-     * will try, at least.
-     */
-#ifdef MBED_CONF_NSAPI_DEFAULT_CELLULAR_APN
-#ifndef MBED_CONF_NSAPI_DEFAULT_CELLULAR_USERNAME
-#define MBED_CONF_NSAPI_DEFAULT_CELLULAR_USERNAME NULL
-#endif
-#ifndef MBED_CONF_NSAPI_DEFAULT_CELLULAR_PASSWORD
-#define MBED_CONF_NSAPI_DEFAULT_CELLULAR_PASSWORD NULL
-#endif
-    cellular->set_credentials(MBED_CONF_NSAPI_DEFAULT_CELLULAR_APN, MBED_CONF_NSAPI_DEFAULT_CELLULAR_USERNAME, MBED_CONF_NSAPI_DEFAULT_CELLULAR_PASSWORD);
-#endif
-#ifdef MBED_CONF_NSAPI_DEFAULT_CELLULAR_SIM_PIN
-    cellular->set_sim_pin(MBED_CONF_NSAPI_DEFAULT_CELLULAR_SIM_PIN);
-#endif
+    return CellularBase::get_default_instance();
```

Comment:
The original code followed the guidelines set out in NetworkInterface.h - for `NetworkInterface::get_target_default_instance`:

> The interface returned should be already configured for use such that its `connect()` method works with no parameters. For connection types needing configuration, settings should normally be obtained from JSON

But the intent was that if an app specifically ask for a `CellularBase`, or any other more specific type, they are aware of the API and expected to configure it themselves. So it's only this call that is expected to do the default lookup from "nsapi.default_xxx", not all the sub-type providers, so the config goes here.

@SeppoTakalo - is that model still working for you?

Reply:
If putting this back, remember it now needs PLMN

Reply:
changed back to original and added plmn setting